plexpy/notifiers.py

```diff
@@ -1574,3 +1614,224 @@ def return_config_options(self):
                          ]
 
         return config_option
+
+
+class Scripts(object):
+
+    def __init__(self, **kwargs):
+        pass
+
+    def conf(self, options):
+            return cherrypy.config['config'].get('Scripts', options)
+
+    def updateLibrary(self):
+        # For uniformity reasons not removed
+        return
+
+    def test(self, subject, message, *args, **kwargs):
+        self.notify(subject, message, *args, **kwargs)
+        return
+
+    def list_scripts(self):
+        scriptdir = plexpy.CONFIG.SCRIPTS_FOLDER
+        scripts = {' ': ''}
+
+        if scriptdir and not os.path.exists(scriptdir):
+            os.makedirs(scriptdir)
+
+        for root, dirs, files in os.walk(scriptdir):
+            for f in files:
+                name, ext = os.path.splitext(f)
+                ext = ext[1:]
+                if ext in ('rb', 'pl', 'bat', 'py', 'sh', 'cmd', 'php'):
+                    fp = os.path.join(root, f)
+                    scripts[fp] = fp
```

Comment:
The dict key will be the full path (i.e. `C:\PlexPy2\PlexPy.py`) and the dict value will be the relative path (i.e. `.\PlexPy.py`).

This way the script selection will show the relative path. You don't need to show the full path in the selection because it's already specified at the top.
![](http://i.imgur.com/I7fhV4F.png)